lib/resources/host.rb

```diff
@@ -48,6 +48,8 @@ def initialize(hostname, params = {})
         @host_provider = LinuxHostProvider.new(inspec)
       elsif inspec.os.windows?
         @host_provider = WindowsHostProvider.new(inspec)
+      elsif inspec.os.darwin?
```

Comment:
We should add a check for the `proto` and skip the resource if udp is used, since we do not support if for darwin. Same should be true for Linux but this is not related to this PR.